runhouse/rns/folders/gcs_folder.py

```diff
@@ -48,7 +48,7 @@ def upload(self, src: str, region: Optional[str] = None):
         )
```

Comment:
Part of me wonders if upload should be private too? Not clear when I'd use `upload` vs. `to`.

Reply:
Oh actually - `upload` means "this folder's system is already set to gcs, but the data is not necessarily there yet, so upload new local data there please." Essentially if a user creates a GCS Folder with dryrun or has an existing GCS folder but wants to put new data in it from a local path, they could use this command. But they could also use `to` to the same effect, but with the extra step of creating a local folder first to send to GCS. I think this is a confusing overlap so I'd be in favor of making this private until we see a real use case. Also worth noting that we already have `.put` to mean "I have a folder, please put this data into it", which also covers a big part of the use case surface here.

Cc @jlewitt1

Reply:
do you think `download` should also be made into a private function then? does `.to('here')` essentially cover its use case?

Reply:
Agreed there are a bunch of methods that more or less do the same thing, I don't see a great use case for it now that we have `to` and `put`. We only use it in one place when copying a folder to a data store:`new_folder.upload(src=local_folder_path)`, but we could just as easily use `.to` instead. 

I think making private for now makes sense. 

@carolineechen also agree re: `download`